edns.go

```diff
@@ -214,9 +216,8 @@ type EDNS0 interface {
 	Option() uint16
 	// pack returns the bytes of the option data.
 	pack() ([]byte, error)
-	// unpack sets the data as found in the buffer. Is also sets
-	// the length of the slice as the length of the option data.
-	unpack([]byte) error
+	// unpack sets the data as found in the buffer.
+	unpack([]byte) bool
```

Comment:
i don't know about this.... two things
1. usually booleans turn into something that have more state
2. regardless of how we error, they (in some cases) do case some useful hints 

And maybe (3): should we really attempt this in this PR?